protocol/swap/types.go

```diff
@@ -0,0 +1,60 @@
+package swap
+
+import (
+	"github.com/athanorlabs/atomic-swap/common/types"
+)
+
+type (
+	Status = types.Status //nolint:revive
+)
+
+// Info contains the details of the swap as well as its status.
+type Info struct {
+	ID             types.Hash // swap offer ID
+	Provides       types.ProvidesCoin
+	ProvidedAmount float64
+	ReceivedAmount float64
+	ExchangeRate   types.ExchangeRate
+	EthAsset       types.EthAsset
+	Status         Status
+	statusCh       <-chan types.Status `json:"-"`
+}
```

Comment:
We need some custom marshaling on this data structure like we did for the Offer type so that, for example, the ID and EthAsset show up in hex instead of [0,1,...31] syntax. I can help out with that if you need it.

Reply:
updated, adding custom marshalling for `Hash` and `EthAsset`. was able to remove the marshalling for `Offer`, since those were the only two types that were custom marshalled